fix: tagging for db, external_table, schema

sfc-gh-swinkler opened the pull request:
tagging for database, external_table and schema were not working. this is caused by two separate problems 1) the command for updating tags never runs and 2) the command itself was incorrect. This PR aims to fix both of these issues. In addition, tags were previously marked as force new, when they really shouldn't be. It would be very bad to accidentally delete a database just because the tags are getting changed, for example. A consequence of removing force new means that we now have to implement update command for external_table, which uses tags, but does not have its own update command. 

## Test Plan
* [ ] acceptance tests

## References

*

alldoami commented on pkg/resources/resource.go:
```diff
@@ -34,15 +35,13 @@ func CreateResource(
 				case schema.TypeInt:
 					valInt := val.(int)
 					qb.SetInt(field, valInt)
-				case schema.TypeList:
```
Why are you getting rid of this?

sfc-gh-swinkler replied:
all that does is type conversion, but it didnt actually work, as it wasn't converting the types properly. we already have a function getTags() that does work. so using that instead

alldoami commented:
/ok-to-test sha=3e8b1cf

alldoami commented:
Could you add tests to test your new update functions?

sfc-gh-swinkler commented:
> Could you add tests to test your new update functions?

sure i will work on this

sfc-gh-swinkler commented:
i added a test case for update and also simplified the update to only handle tags, as that is the purpose of this PR. leaving force new for all other attributes, as that should be handled in a separate pr.

alldoami commented:
/ok-to-test sha=dffc3ac
